src/core/client_channel/load_balanced_call_destination.cc

```diff
@@ -0,0 +1,334 @@
+}
+
+}  // namespace
+
+void LoadBalancedCallDestination::StartCall(
+    UnstartedCallHandler unstarted_handler) {
+  // If there is a call tracer, create a call attempt tracer.
+  bool* is_transparent_retry_metadata =
+      unstarted_handler.UnprocessedClientInitialMetadata().get_pointer(
+          IsTransparentRetry());
+  bool is_transparent_retry = is_transparent_retry_metadata != nullptr
+                                  ? *is_transparent_retry_metadata
+                                  : false;
+  MaybeCreateCallAttemptTracer(is_transparent_retry);
+  // Spawn a promise to do the LB pick.
+  // This will eventually start the call.
+  unstarted_handler.SpawnGuardedUntilCallCompletes(
+      "lb_pick", [unstarted_handler, picker = picker_]() mutable {
+        return Map(
+            // Wait for the LB picker.
+            CheckDelayed(Loop(
+                [last_picker =
+                     RefCountedPtr<LoadBalancingPolicy::SubchannelPicker>(),
+                 unstarted_handler, picker]() mutable {
+                  return Map(
+                      picker.Next(last_picker),
+                      [unstarted_handler, &last_picker](
+                          RefCountedPtr<LoadBalancingPolicy::SubchannelPicker>
+                              picker) mutable {
+                        last_picker = std::move(picker);
+                        // Returns 3 possible things:
+                        // - Continue to queue the pick
+                        // - non-OK status to fail the pick
+                        // - a connected subchannel to complete the pick
+                        return PickSubchannel(*last_picker, unstarted_handler);
+                      });
+                })),
+            // Create call stack on the connected subchannel.
+            [unstarted_handler](
+                std::tuple<
+                    absl::StatusOr<RefCountedPtr<UnstartedCallDestination>>,
+                    bool>
+                    pick_result) {
+              auto& connected_subchannel = std::get<0>(pick_result);
+              const bool was_queued = std::get<1>(pick_result);
+              if (!connected_subchannel.ok()) {
+                return connected_subchannel.status();
+              }
+              // LB pick is done, so indicate that we've committed.
+              auto* on_commit = MaybeGetContext<LbOnCommit>();
+              if (on_commit != nullptr && *on_commit != nullptr) {
+                (*on_commit)();
+              }
+              // If it was queued, add a trace annotation.
+              if (was_queued) {
+                auto* tracer =
+                    MaybeGetContext<ClientCallTracer::CallAttemptTracer>();
+                if (tracer != nullptr) {
+                  tracer->RecordAnnotation("Delayed LB pick complete.");
+                }
+              }
+              // Delegate to connected subchannel.
+              // FIXME: need to insert LbCallTracingFilter at the top of the
```

Comment:
I think we should do this as part of this PR.

Reply:
going to suggest a follow-on -- better to get the chain that's piled up submitting (since this part isn't required for the urgent work ahead)

Reply:
(it'll likely happen this week)